sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_discovery.py

```diff
@@ -0,0 +1,102 @@
+# ------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+
+import os
+import time
+import dns.resolver
+from dns.resolver import NXDOMAIN, LifetimeTimeout, NoNameservers  # cspell:disable-line
+from dns.rdatatype import SRV  # cspell:disable-line
+from ._constants import DISABLE_APPCONFIGURATION_DISCOVERY, HTTPS_PREFIX
+
+
+class SRVRecord:
+    priority: int
+    weight: int
+    port: int
+    target: str
+    protocol: str = HTTPS_PREFIX
+
+    def __init__(self, answer):
+        self.priority = answer.priority
+        self.weight = answer.weight
+        self.port = answer.port
+        self.target = str(answer.target).rstrip(".")
+
+
+def find_auto_failover_endpoints(endpoint: str):
+    if os.environ.get(DISABLE_APPCONFIGURATION_DISCOVERY, "").lower() == "true":
+        return []
+    known_domain = _get_known_domains(endpoint)
+    if known_domain is None:
+        return []
+
+    origin = _find_origin(endpoint)
+    if origin is None or not _validate(known_domain, origin.target):
```

Comment:
the `_validate` method seem to guard against logic errors in the program and silently hiding them. For example, under what circumstances would `origin.target` not be an instance of `str`? And why did it make it this far? Adding type annotations and running a static type checker can help you catch these kind if "check if this thing that can't happen happened because of an issue somewhere earlier in my code but don't raise an exception and instead skip logic" error checks.

Reply:
I've cleaned this up a bit. at this point origin is just the value returned by `dns.resolver`. This is us validating that uri is as expected.